include/librepr/enum/accessor.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+#include <string>
+#include <type_traits>
+#include <array>
+#include <string_view>
+#include <algorithm>
+
+#include <librepr/util/type_list.h>
+
+#include "util.h"
+
+namespace librepr::ctei {
+
+template <typename T, EnumKind Kind, typename Structure = TypeList<>>
+  requires std::is_enum_v<T>
+struct Accessor {
+  using structure            = Structure;
+  using underlying           = std::underlying_type_t<T>;
+  constexpr static auto kind = Kind;
+
+private:
+  static consteval auto get_enum_names() {
+    if constexpr (structure::size == 0) {
+      return std::array<std::string_view, 0>{};
+    } else {
+      auto constexpr full_size = structure::invoke([]<typename... Range>() { return ((Range::size) + ...); });
```

Comment:
Should probably make a metafunction that does this and call ``get_enum_names`` with it